### PR TITLE
Sign up

### PR DESCRIPTION
**Pull request by anhkhang2304**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+class UsersController < ApplicationController
+  def new
+  	@user = User.new
```
Comment (HoangThiNhung) [inline app/controllers/users_controller.rb]: indent

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  end
+
+  def show
+  	@user = User.find_by id: params[:id]
```
Comment (HoangThiNhung) [inline app/controllers/users_controller.rb]: indent

File: app/helpers/users_helper.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+module UsersHelper
+  def gravatar_for(user, size: 80)
```
Comment (HoangThiNhung) [inline app/helpers/users_helper.rb]: remove ()

File: app/helpers/users_helper.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+    gravatar_id = Digest::MD5::hexdigest(user.email.downcase)
```
Comment (HoangThiNhung) [inline app/helpers/users_helper.rb]: remove ()

File: app/models/user.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  validates :name, presence: true, length: {maximum: 50}
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: {maximum: 250},
+  	format: {with: VALID_EMAIL_REGEX},
```
Comment (HoangThiNhung) [inline app/models/user.rb]: indent

File: app/models/user.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+  	uniqueness: true
```
Comment (HoangThiNhung) [inline app/models/user.rb]: indent

File: db/migrate/20160817034629_add_index_to_users_email.rb [hunk 7]
```diff
@@ -0,0 +1,5 @@
+class AddIndexToUsersEmail < ActiveRecord::Migration[5.0]
+  def change
+  	  add_index :users, :email, unique: true
```
Comment (HoangThiNhung) [inline db/migrate/20160817034629_add_index_to_users_email.rb]: indent

File: app/controllers/users_controller.rb [hunk 8]
```diff
@@ -1,4 +1,26 @@
...
+    @user = User.find_by id: params[:id]
+  end
+
+  def create
+    @user = User.new user_params
+    if @user.save
+      flash[:success] = "Register successful!"
+      redirect_to @user
+    else
+      render "new"
```
Comment (thangtx88) [inline app/controllers/users_controller.rb]: render :new

**Pull request merged**